src/content/docs/en/recipes/dynamically-importing-images.mdx

```diff
@@ -0,0 +1,96 @@
+---
+title: Dynamically Import Images
+description: Learn how to dynamically import images using Vite's import.meta.glob function
+i18nReady: true
+type: recipe
+---
+import FileTree from '~/components/FileTree.astro'
+
+When working with custom components that include images, there will be times where you will need to dynamically import the image paths. But if the images are located in the `src` directory, it is possible to run into Astro or Vite import errors.  
+
+In this recipe, you will learn how to dynamically import your images the correct way using Vite's `import.meta.glob` function. 
+
+## Recipe
+
+1. Create a new `assets` folder under the `src` directory and add your images inside that new folder. 
+
+<FileTree>
+- src/
+  - assets/
+    - example-image.jpg
+    - example-image-2.jpg
+    - example-image-3.jpg
+</FileTree>
+
+2. Create a new Astro component and import the `<Image>` component.
+
+    ```astro title="src/components/MyCustomCardComponent.astro" 
+    ---
+     import { Image } from 'astro:assets';
+    ---
+    ```
+
+3. Create the `props` for your component.
+
+    ```astro title="src/components/MyCustomCardComponent.astro" ins={3}
+    ---
+     import { Image } from 'astro:assets';
+     const { imagePath, altText, name, age } = Astro.props;
+    ---
+    ``` 
+
+4. Create a new `images` variable and use the `import.meta.glob` function which returns an object of all of the image paths inside the `assets` folder. 
+
+    ```astro title="src/components/MyCustomCardComponent.astro" ins={4}
+    ---
+     import { Image } from 'astro:assets';
+     const { imagePath, altText, name, age } = Astro.props;
+     const images = import.meta.glob("../assets/*");
```

Comment:
When using dynamic images in several locations, it can be a bit tedious to write this in several locations with different relatives path (eg `../assets/*` and `../../assets/*`). It could be great to show a more centralized pattern as well as shown on Discord:
```ts
// file located in a more central place
async function imp(filePath) {
  const images = import.meta.glob('/src/**/*.{jpeg,jpg,png,gif}')
  const keys = Object.keys(images)
  const key = keys.find(key => key.includes(filePath))

  if (!key) throw new Error(`Image ${filePath} not found`)

  const image = await images[key]()
  return image.default
}

const a = await imp('/assets/home/hero.jpg')
console.log(a)
```
Also, it may be great to restrict the extension types like in the code snippet above to improve TS typings